### DISCUSSION TITLE
How to let IM adopt suitable system-resource-values from current system, when IM binaries were previously moved from another system

### DISCUSSION THREAD

**Discussion by motz17** [ImageMagick/ImageMagick#6223] [General]
Hi!

I have following scenario:

**1.)** I run a docker-instance of Ubuntu 22.04, where I install ImageMagick 7.1.0-46 Q16-HDRI x86_64

I can see, that certain resources are set:
> convert -list resource
> Resource limits:
>   Width: 461.169PP
>   Height: 461.169PP
>   Area: 33.5734GP
>   List length: unlimited
>   Memory: 1024000000B
>   Map: 1024000000B
>   Disk: 40000000000B
>   File: 786432
>   Thread: 4
>   Throttle: 0
>   Time: unlimited

I'd assume, that some of those values are fetched from the current system (like File, Thread, Memory).

**2.)** Now I'm transferring & running this docker image (with the installed IM on it) on other systems (physically different hardware). These other systems might be much more powerful / or much less powerful (CPU cores, memory, disk-size,..)  than the original system where I installed IM.

I guess, I could manually change those values in the policy.xml, like opting in: 

**3.)** => My question: Is there any way to let IM auto'magick'ally detect /recalibrate the system resources after the installation?

Thx in advance! 
Best regards
Martin

 
 
 

_sidenote: this is the output of magick --version_

> Version: ImageMagick 7.1.0-46 Q16-HDRI x86_64 5ef3d4d66:20220816 https://imagemagick.org
> Copyright: (C) 1999 ImageMagick Studio LLC
> License: https://imagemagick.org/script/license.php
> Features: Cipher DPC HDRI Modules OpenMP(4.5)
> Delegates (built-in): bzlib cairo djvu fftw fontconfig freetype gslib gvc heic jbig jng jp2 jpeg jxl lcms lqr ltdl lzma openexr pangocairo png ps raqm raw rsvg tiff webp wmf x xml zip zlib
> Compiler: gcc (11.3)

**Comment by snibgo** [comment]
Check your 'policy.xml'. Does that set any limits? If it does, then I expect a shell script could be written to change the limits to values more suitable for a given system.

If policy.xml doesn't set limits, then I think the limits you get are set at execution time, not installation time.
